plugins/controlboard/src/ControlBoardDriver.cpp

```diff
@@ -404,9 +433,15 @@ bool GazeboYarpControlBoardDriver::sendPositionsToGazebo(Vector &refs)
 bool GazeboYarpControlBoardDriver::sendPositionToGazebo(int j,double ref)
 {
     gazebo::msgs::JointCmd j_cmd;
-    prepareJointMsg(j_cmd,j,ref);
-    m_jointCommandPublisher->WaitForConnection();
-    m_jointCommandPublisher->Publish(j_cmd);
+
+    if(ref != m_oldReferencePositions[j])
+    {
+        // std::cout << "Sending new command: new ref is " << ref << "; old ref was " << m_oldReferencePositions[j] << std::endl;
```

Comment:
I understand that this is subjective, but I find that if we keep all the old debug messages in the code the code will become quickly a unreadable mess of commented out prints... 

However given that this is just my personal taste, no problem!